### PR TITLE
[fix][test]Fix cannot resolve method addLast error

### PR DESCRIPTION
**Pull request by gaozhangmin**
### Motivation
Cannot resolve method 'addLast(java.lang.String, org.apache.pulsar.client.api.MockBrokerService.MockServerCnx)' 

### Modifications
Cast `MockServerCnx` to ChannelHandler

### Documentation

Check the box below or label this PR directly.

Need to update docs? 

- [ ] `doc-required` 
(Your PR needs to update docs and you will update later)
  
- [x] `no-need-doc` 
(Please explain why)
  
- [ ] `doc` 
(Your PR contains doc changes)

- [ ] `doc-added`
(Docs have been already added)

### CODE REVIEW

**Comment by nodece** [comment]
I notice the Pulsar have many similar codes, could you fix these code?

It should be working, but the IDEA will tell us there are errror here.

**Comment by gaozhangmin** [comment]
> I notice the Pulsar have many similar codes, could you fix these code?
> 
> It should be working, but the IDEA will tell us there are error here.

I only find one of this kind error.  @nodece

**Comment by gaozhangmin** [comment]
@nodece PTAL again, Thanks for pointing it out

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/PulsarChannelInitializer.java [hunk 1]
```diff
@@ -144,7 +145,7 @@ protected void initChannel(SocketChannel ch) throws Exception {
         // auto-read.
         ch.pipeline().addLast("flowController", new FlowControlHandler());
         ServerCnx cnx = newServerCnx(pulsar, listenerName);
-        ch.pipeline().addLast("handler", cnx);
+        ch.pipeline().addLast("handler", (ChannelHandler) cnx);
```
Comment (Jason918) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/service/PulsarChannelInitializer.java:148]: Sorry, I didn't get this. `ServerCnx` extends `ChannelInboundHandlerAdapter`, why is this necessary?

Comment (gaozhangmin) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/service/PulsarChannelInitializer.java:148]: Solve the error hints by the IDEA. @Jason918

Comment (Jason918) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/service/PulsarChannelInitializer.java:148]: image

No errors in my local IDEA, my version: IntelliJ IDEA 2021.2.4 (Ultimate Edition)

Comment (michaeljmarshall) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/service/PulsarChannelInitializer.java:148]: @gaozhangmin - are you able to provide a screenshot of the error message? My IntelliJ does not report this error, either.

Comment (nodece) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/service/PulsarChannelInitializer.java:148]: image


I have the same issue, this is my screenshot on IntelliJ IDEA 2022.1 (Ultimate Edition)

Comment (michaeljmarshall) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/service/PulsarChannelInitializer.java:148]: Thanks for sharing @nodece. I am running with IntelliJ IDEA 2021.3.3 (Ultimate Edition). I wonder why we're seeing this discrepancy.

Comment (gaozhangmin) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/service/PulsarChannelInitializer.java:148]: I also use  IntelliJ IDEA 2022.1  @michaeljmarshall  @nodece

**Comment by BewareMyPower** [comment]
I think it's a issue of Intellij Idea, there is no need to add a cast in code.

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/PulsarChannelInitializer.java (hunk 1 above)
Comment (BewareMyPower) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/service/PulsarChannelInitializer.java:148]: image

It looks like a bug of Idea, both `PulsarDecoder` and my customized `Decoder` extend the same class `ChannelInboundHandlerAdapter`, while inheriting `Decoder` works well.

**Comment by nodece** [comment]
Clone the Pulsar repo to a new folder, then use the IDEA to open this new project, it is working.
